lib/ddtrace/contrib/elasticsearch/core.rb

```diff
@@ -0,0 +1,61 @@
+require 'uri'
+require 'ddtrace/pin'
+require 'ddtrace/ext/app_types'
+
+URL = 'elasticsearch.url'.freeze
+METHOD = 'elasticsearch.method'.freeze
+TOOK = 'elasticsearch.took'.freeze
+PARAMS = 'elasticsearch.params'.freeze
+BODY = 'elasticsearch.body'.freeze
+
+DEFAULTSERVICE = 'elasticsearch'.freeze
+
+module Datadog
+  module Contrib
+    module Elasticsearch
+      # Elastic Search integration.
+      module TracedClient
+        def initialize(*args)
```

Comment:
Thanks for reviewing! Indeed I was quite surprised at first too. This has the advantage of being exactly the way our Python integration does it. One advantage is that it gives a of control to the user, who quite never needs to subclass or patch anything. Just pin.onto and pin.get_from on your object, and you're fine. This might make more sense now that I have an ES implementation actually using it. Eg https://github.com/DataDog/dd-trace-rb/pull/39/files#diff-e80da4e7288be738cb0e5c1107db299dR77